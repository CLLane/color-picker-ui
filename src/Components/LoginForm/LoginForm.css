.sign-in_form {
  min-height: 30vh;
  display: flex;
  padding-left: 33%;
  padding-right: 33%;
  flex-direction: column;
  justify-content: space-evenly;
}

button {
    border: none;
    border-radius: 10px;
    height: 43px;
    background: rgba(167, 241, 253, 0.95);
  }


input {
    border: none;
    border-radius: 10px;
    height: 45px;
    background: rgba(255, 255, 255, 0.95);
    text-align: center;
}