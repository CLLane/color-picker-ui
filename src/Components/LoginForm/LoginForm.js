import React, { Component } from 'react';
import './LoginForm.css';

export class LoginForm extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      newUser: false
    }
  }
  
  handleChange = e => {
    const { clearError } = this.props;
    clearError();
    this.setState({ [e.target.name]: e.target.value });
  };

  clearInputs = () => {
    this.setState({
      email: '',
      password: ''
    });
  };
  
  handleLogin = e => {
    const { loginUser } = this.props;
    const { email, password } = this.state;
    e.preventDefault();
    loginUser(email, password);
    this.clearInputs();
  };

  toggleNewUser = e => {
    e.preventDefault();
    const { newUser } = this.state;
    this.setState({ newUser: !newUser });
  };

  handleSignUp = e => {
    const { signUpUser } = this.props;
    const { email, password } = this.state;
    e.preventDefault();
    signUpUser(email, password);
    this.clearInputs();
  };

  render() {
    const { email, password, newUser } = this.state;
    const { error } = this.props;
    return (
      <form className='sign-in_form'>
        <input type='text' value={email} placeholder='Email' name='email' onChange={this.handleChange}></input>
        <input type='password' value={password} placeholder='Password' name='password' onChange={this.handleChange}></input>
        { !newUser && <button onClick={this.handleLogin}>{<img src="https://fontmeme.com/permalink/191011/8b36e600f13d2e8bca6002973f735b84.png" alt="graffiti-fonts" border="0"/>}</button> }
        {newUser && <button onClick={this.handleSignUp}>{<img src="https://fontmeme.com/permalink/191011/8e4934228d7a5e0d23888f81c63bcfcd.png" alt="graffiti-fonts" border="0"/>}</button> }
        {newUser && <button onClick={this.toggleNewUser}>{<img src="https://fontmeme.com/permalink/191011/638e3f86d1ed72f908e3e5c26e09a2f5.png" alt="Sign In" border="0"/>}</button>}
        { error && <p>Login failed, please re-enter information.</p> }
        {!newUser && <button onClick={this.toggleNewUser}>{<img src="https://fontmeme.com/permalink/191011/1857f34f8065b95bfa5d7c167f12c23d.png" alt="graffiti-fonts" border="0"/>}</button>}
      </form>
    );
  };
};

export default LoginForm;