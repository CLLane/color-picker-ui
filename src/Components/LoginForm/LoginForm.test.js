import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';
import './LoginForm.css';

describe('LoginForm', () => {
  let wrapper;
  let error;
  let mockClearError;
  let mockLoginUser;
  beforeEach(() => {
    mockClearError = jest.fn();
    mockLoginUser = jest.fn();
    wrapper = shallow(<LoginForm clearError={mockClearError} loginUser={mockLoginUser} />);
  });

  it('should match the snapshot that allows an existing user to login', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot that allows a new user to sign up', () => {
    wrapper.setState({ newUser: true });
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when handle change is called', () => {
    const mockEvent = {
      target: {
        name: 'email', 
        value: 'bob@gmail.com'
      }
    };
    wrapper.instance().handleChange(mockEvent);
    const expectedEmail = 'bob@gmail.com';
    expect(mockClearError).toHaveBeenCalled();
    expect(wrapper.state('email')).toEqual(expectedEmail);
  });

  it('should clear input values when called', () => {
    wrapper.setState({email: 'hi@gmailcom', password: 12345 });
    wrapper.instance().clearInputs();
    expect(wrapper.state('email')).toEqual('');
    expect(wrapper.state('password')).toEqual('');
  });

  it('should call login user with the proper email and password, and then clear inputs', () => {
    const mockEvent = {
      preventDefault: jest.fn()
    };
    wrapper.setState({email: 'bob@gmail.com', password: 12345 });
    wrapper.instance().handleLogin(mockEvent);
    expect(mockLoginUser).toHaveBeenCalledWith('bob@gmail.com', 12345);
  });

  it('it should toggle between showing a for a new or existing user', () => {
    const mockEvent = {
      preventDefault: jest.fn()
    };
    wrapper.instance().toggleNewUser(mockEvent);
    expect(wrapper.state('newUser')).toEqual(true);
  });

});