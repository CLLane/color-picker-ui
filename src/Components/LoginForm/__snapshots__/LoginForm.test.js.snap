// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot that allows a new user to sign up 1`] = `
<form
  className="sign-in_form"
>
  <input
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />
  <input
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <button
    onClick={[Function]}
  >
    <img
      alt="graffiti-fonts"
      border="0"
      src="https://fontmeme.com/permalink/191011/8e4934228d7a5e0d23888f81c63bcfcd.png"
    />
  </button>
  <button
    onClick={[Function]}
  >
    <img
      alt="Sign In"
      border="0"
      src="https://fontmeme.com/permalink/191011/638e3f86d1ed72f908e3e5c26e09a2f5.png"
    />
  </button>
</form>
`;

exports[`LoginForm should match the snapshot that allows an existing user to login 1`] = `
<form
  className="sign-in_form"
>
  <input
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />
  <input
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <button
    onClick={[Function]}
  >
    <img
      alt="graffiti-fonts"
      border="0"
      src="https://fontmeme.com/permalink/191011/8b36e600f13d2e8bca6002973f735b84.png"
    />
  </button>
  <button
    onClick={[Function]}
  >
    <img
      alt="graffiti-fonts"
      border="0"
      src="https://fontmeme.com/permalink/191011/1857f34f8065b95bfa5d7c167f12c23d.png"
    />
  </button>
</form>
`;
